float32 disp_angle
float32 point_pos